docs/zh_CN/extension/native/develop/plugins_tutorial.md

```diff
@@ -185,54 +185,49 @@ require (
-    2. 由于 Go 语言插件系统对依赖的路径有非常严格的要求，为了确保插件可以顺利运行，建议在 eKuiper 主项目里进行编译。在 eKuiper 项目里，添加如下名为 sample.mod 的文件并让它指向真正的插件项目，以便于插件编译。如果你的eKuiper与插件项目在同一目录，这里有个例子：
-       ```
-       module github.com/lf-edge/ekuiper
-       go 1.18
-       require samplePlugin v0.0.0
-       replace samplePlugin => ../samplePlugin   # replace samplePlugin with local samplePlugin module
-       ```
-    3. 经过这些配置，你的插件项目与eKuiper项目目录结构应该是这样
-       ``` 
+3. 编译环境设置
+    1. 用户可以把 eKuiper 和 插件放在同一目录下，项目目录类似于如下结构：
+      ```
+      workspace
         ekuiper
-          sample.mod         //new added sample.mod file in step2
-          go.mod             //existing ekuiper default mod file
-          extensions.mod     //existing extensions mod file for plugins
+          go.mod
         samplePlugin
-          go.mod             //new plugin project default mod file
-       `
+          go.mod
+      ```
+    2. 在1.9.0 里，我们使用了 go workspace 功能重构了子模块的go mod构建方式，所以我们可以使用 go workspace 解决依赖问题。进入 workspace 目录里，创建工作区：  
+      ```shell
+      go work init ./ekuiper ./samplePlugin
+      ```
+    3. 经过这些配置，你的插件项目与eKuiper项目目录结构应该是这样
+      ``` 
+        workspace
+          ekuiper
+            go.mod             
+          samplePlugin
+            go.mod
+          go.work
+      ```
        
-5. 在 eKuiper 目录下，编译插件和eKuiper
+4. 在 workspace 目录下，编译插件和eKuiper
    ```shell
     # compile the eKuiper
-    go build -trimpath -o $eKuiperPath/_build/$build/bin/kuiperd cmd/kuiperd/main.go
+    go build -trimpath -o ./_build/$build/bin/kuiperd cmd/kuiperd/main.go
 
+    cd $workspacePath
     # compile the plugin that using self-managed project within eKuiper project
-    go build -trimpath -modfile sample.mod --buildmode=plugin -o $eKuiperPath/_build/$build/plugins/sinks/Mysql@v1.0.0.so ../samplePlugin/sinks/mysql.go
+    go build -trimpath --buildmode=plugin -o ./ekuiper/_build/$build/plugins/sinks/Mysql@v1.0.0.so ./samplePlugin/sinks/mysql.go
    ```
+**注意**：插件命名有限制，详见[插件总览](../overview.md)。
 
 #### Docker编译
 
-eKuiper 提供了开发版本 docker 镜像。从 1.7.1 开始，开发镜像为 x.x.x-dev (0.4.0 到 1.7.0 之间版本的开发镜像为x.x.x，例如`lfedge/ekuiper:0.4.0`。)；与运行版本相比，开发版提供了 go 开发环境，使得用户可以在编译出在 eKuiper 正式发布版本中完全兼容的插件。Docker 中编译步骤如下：
-1. 运行 eKuiper 开发版本 docker。需要把本地插件目录 mount 到 docker 里的目录中，这样才能在 docker 中访问插件项目并编译。笔者的插件项目位于本地 `/var/git` 目录。下面的命令中，我们把本地的 `/var/git`目录映射到 docker 内的 `/home` 目录中。
+eKuiper 提供了开发版本 docker 镜像。从 1.7.1 开始，开发镜像为 x.x.x-dev (0.4.0 到 1.7.0 之间版本的开发镜像为x.x.x，例如`lfedge/ekuiper:0.4.0`。)；与运行版本相比，开发版提供了 go 开发环境，使得用户可以在编译出在 eKuiper 正式发布版本中完全兼容的插件。由于1.9.0版本之后才使用go workspace功能，所以后面的步骤只适用于1.9.0之后的版本。在 Docker 中编译步骤如下：
+1. 运行 eKuiper 开发版本 docker。需要把本地插件目录 mount 到 docker 里的目录中，这样才能在 docker 中访问插件项目并编译。笔者的插件项目位于本地 `/var/git` 目录。下面的命令中，我们把本地的 `/var/git`目录映射到 docker 内的 `/go/plugins` 目录中。
     ```go
-    docker run -d --name kuiper-dev --mount type=bind,source=/var/git,target=/home lfedge/ekuiper:1.3.0
+    docker run -d --name kuiper-dev --mount type=bind,source=/var/git,target=/go/plugins lfedge/ekuiper:1.9.0-alpha.0-dev
```

Comment:
Suggest to use a non-specific version number here, at least, not an alpha version. Like 1.x.x-dev